evalml/automl/engine/cf_engine.py

```diff
@@ -82,11 +84,17 @@ def is_cancelled(self):
 class CFEngine(EngineBase):
     """The concurrent.futures (CF) engine"""
 
-    def __init__(self, client):
-        if not isinstance(client, CFClient):
+    def __init__(self, client=None):
+        """
+        Args:
+            client (None or CFClient): If None, creates a threaded pool for processing.  Defaults to None.
+        """
+        if client is not None and not isinstance(client, CFClient):
             raise TypeError(
                 f"Expected evalml.automl.engine.cf_engine.CFClient, received {type(client)}"
             )
+        elif client is None:
+            client = CFClient(ThreadPoolExecutor())
```

Comment:
The issue lists doing this for the dask engine as a proposed change. Is that no longer the case? I think we should do this for the dask engine as well or not do it for either.

Reply:
Done!